### PR TITLE
Recipe for libflint

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: recipes/flint/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+S{% set version = "2.5.3.dev0" %}
+{% set commit = "6e1c767c29b5d1c0e94d33512181ad63e73f5fc7" %}
```
Comment (isuruf) [inline recipes/flint/meta.yaml]: I've use the git version because all releases are GPL 2 and the development version is LGPL 2

Comment (jakirkham) [inline recipes/flint/meta.yaml]: Could you please raise an issue with the dev to get a release made?

Comment (isuruf) [inline recipes/flint/meta.yaml]: Will do. I have some fixes for windows as a PR as well.

**Comment by jakirkham** [comment]
Unfortunately, we need to discuss naming here. There is the C library presented here called "FLINT". There is a Python package that we will inevitably include called "flint". There is also a Python library that interfaces with FLINT called "python-flint". Just to hopefully speed up this discussion and not bikeshed this forever, I'd like to propose we name this "libflint" and the other two packages (when made) will keep their names as is. Thoughts?

cc @pelson @ocefpaf @msarahan @mcg1969

**Comment by mcg1969** [comment]
I think your reasoning is sound, although since the web site for this package is http://www.flintlib.org, I would suggest naming this package `flintlib`.

**Comment by mcg1969** [comment]
Or perhaps it should be `flint2`, since that's the name of the GitHub repo: https://github.com/wbhart/flint2

**Comment by jakirkham** [comment]
Sure, either sound reasonable to me.

**Comment by isuruf** [comment]
@jakirkham, I've often had to pass directories where header files are located. How about adding `-I$PREFIX/include` to `CFLAGS` in toolchain and same for libs ?

**Comment by jakirkham** [comment]
Yep, I think that is reasonable. Would you like to do a PR? 😄

**Comment by isuruf** [comment]
Will do

**Comment by jakirkham** [comment]
Since we can't run the tests on Travis CI (timeout issues), could we have a simple C or C++ program that does a simple test at the end? I know it is less rigorous, but at least we could make sure things are linked ok and usable that way.

**Comment by jakirkham** [comment]
On second thought, I'm not sold on `flint2`. Hard-coding version numbers in packages seems like a deplorable practice IMHO. While that is a trend in Linux package management (where they don't benefit from the lovely environments that we do), even Ubuntu and Arch Linux just went for `flint`. Let's either go `libflint` or `flintlib`. The former already happens with C/C++ libraries. The latter matches the webpage name.

**Comment by isuruf** [comment]
Tests pass now

File: recipes/libflint/build.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+source activate "${CONDA_DEFAULT_ENV}"
+
+chmod +x configure
+
+echo "int flint_test_multiplier(){return 1;}" > test_helpers.c
```
Comment (jakirkham) [inline recipes/libflint/build.sh]: So, I'm guessing this is the test that file that caused this to hang. I'm ok with skipping it and this seems as good of way as any. Though could we please add a comment above explaining what we are doing and why?

File: recipes/libflint/build.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+source activate "${CONDA_DEFAULT_ENV}"
```
Comment (jakirkham) [inline recipes/libflint/build.sh]: We should be able to drop this now as we are using a newer `conda-build` on OS X.

File: recipes/libflint/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{% set version = "2.5.3.dev0" %}
+{% set commit = "6e1c767c29b5d1c0e94d33512181ad63e73f5fc7" %}
+
+package:
+  name: libflint
+  version: {{ version }}
+
+source:
+  fn: {{ commit }}.tar.gz
+  url: https://github.com/wbhart/flint2/archive/{{ commit }}.tar.gz
```
Comment (jakirkham) [inline recipes/libflint/meta.yaml]: This now appears to be present in the non-forked version. Could we please switch to the `fredrik-johansson` user or is there some other reason not to?

File: recipes/libflint/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{% set version = "2.5.3.dev0" %}
+{% set commit = "6e1c767c29b5d1c0e94d33512181ad63e73f5fc7" %}
```
Comment (jakirkham) [inline recipes/libflint/meta.yaml]: I have raised this issue ( https://github.com/wbhart/flint2/issues/275 ) to see if we can get a new release.

File: recipes/libflint/meta.yaml (hunk 4 above)
Comment (isuruf) [inline recipes/libflint/meta.yaml]: `wbhart/flint2` is the main development repo. Github doesn't let you make a forked repo into the main one.

File: recipes/libflint/build.sh (hunk 2 above)
Comment (isuruf) [inline recipes/libflint/build.sh]: I've explained this in a comment. It reduces the number of random tests in flint.

File: recipes/libflint/meta.yaml (hunk 4 above)
Comment (jakirkham) [inline recipes/libflint/meta.yaml]: Ok SGTM. That is what I was missing.

File: recipes/libflint/build.sh (hunk 2 above)
Comment (jakirkham) [inline recipes/libflint/build.sh]: Thanks.

File: recipes/libflint/meta.yaml (hunk 5 above)
Comment (isuruf) [inline recipes/libflint/meta.yaml]: From that thread, it looks like it will take some time to get a new release

**Comment by ocefpaf** [comment]
@isuruf if a release is far away it is OK to merge this. Usually we prefer stable official releases, but we know that not every project follow issue those in a timely maner.

Let me know if this is good to go.

**Comment by jakirkham** [comment]
So the challenging point here is we are creating a release (note this commit is not released not even as dev) to get a different license. The first point of creating a release is what is most concerning IMHO. The second point is the reason this is being done is to obtain different licensing conditions. Not sure if this is a procedure we want to start encouraging. Though maybe others can weigh in.

cc @conda-forge/core

**Comment by patricksnape** [comment]
Please rebase on top of master as an important fix went in for OSX builds.

File: recipes/libflint/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
+{% set version = "2.5.2" %}
+
+package:
+  name: libflint
+  version: {{ version }}
+
+source:
+  fn: flint-{{ version }}.tar.gz
+  url: http://flintlib.org/flint-{{ version }}.tar.gz
+  md5: cda885309362150196aed66a5e0f0383
```
Comment (jakirkham) [inline recipes/libflint/meta.yaml]: `sha256` is preferred.

**Comment by isuruf** [comment]
Tests pass now

File: recipes/libflint/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+test:
+  commands:
+    - test -f $PREFIX/include/flint/flint.h
+    - test -f $PREFIX/lib/libflint.a
+    - test -f $PREFIX/lib/libflint.so         # [linux]
+    - test -f $PREFIX/lib/libflint.dylib      # [osx]
+
+about:
+  home: http://flintlib.org
+  license: GPL v2+
```
Comment (jakirkham) [inline recipes/libflint/meta.yaml]: Done below.

File: recipes/libflint/meta.yaml [hunk 8]
```diff
@@ -32,6 +32,7 @@ test:
 about:
   home: http://flintlib.org
   license: GPL v2+
+  license_file: LICENSE
```
Comment (jakirkham) [inline recipes/libflint/meta.yaml]: Because not everyone nicely names their license file `LICENSE`. 😉

**Comment by jakirkham** [comment]
Any comments from anyone else?

**Pull request merged**